pkg/kubelet/api/services.go

```diff
@@ -23,13 +23,44 @@ import (
 )
 
 // RuntimeService interface should be implemented by a container runtime.
-// The methods should be thread-safe.
+// The methods are thread-safe.
 type RuntimeService interface {
+	ContainerManager
+	PodSandboxManager
+	RuntimeVersioner
+}
+
+// RuntimeVersioner contains methods for runtime name, version and API version.
+type RuntimeVersioner interface {
 	// Version returns the runtime name, runtime version and runtime API version
 	Version(apiVersion string) (*runtimeApi.VersionResponse, error)
+}
+
+// ContainerManager contains methods to manipulate containers managed by a
+// container runtime. The methods are thread-safe.
+type ContainerManager interface {
+	// StartContainer starts the container.
+	StartContainer(containerID string) error
+	// StopContainer stops a running container with a grace period (i.e., timeout).
+	StopContainer(containerID string, timeout int64) error
+	// RemoveContainer removes the container.
+	RemoveContainer(containerID string) error
+	// ListContainers lists all containers by filters.
+	ListContainers(filter *runtimeApi.ContainerFilter) ([]*runtimeApi.Container, error)
+	// ContainerStatus returns the status of the container.
+	ContainerStatus(containerID string) (*runtimeApi.ContainerStatus, error)
+	// Exec executes a command in the container.
+	Exec(containerID string, cmd []string, tty bool, stdin io.Reader, stdout, stderr io.WriteCloser) error
+}
+
+// PodSandboxManager contains methods for operating on PodSandboxes. The methods
+// are thread-safe.
+type PodSandboxManager interface {
 	// CreatePodSandbox creates a pod-level sandbox.
 	// The definition of PodSandbox is at https://github.com/kubernetes/kubernetes/pull/25899
 	CreatePodSandbox(config *runtimeApi.PodSandboxConfig) (string, error)
+	// CreateContainer creates a new container in specified PodSandbox.
+	CreateContainer(podSandboxID string, config *runtimeApi.ContainerConfig, sandboxConfig *runtimeApi.PodSandboxConfig) (string, error)
```

Comment:
The reason is valid, but I still think the method should be included in the ContainerManager instead, since it's part of the container lifecycle management. We can change it later if it becomes obvious in the tests/code that this belongs to the sandbox manager.